modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorTests.java

```diff
@@ -504,4 +518,337 @@ protected void doShutdown() throws IOException {
+
+        // Verify that a nested object was created (classic behavior)
+        assertThat(ingestDocument.hasField("my.target"), is(true));
+        Object target = ingestDocument.getFieldValue("my.target", Object.class);
+        assertThat(target, instanceOf(Map.class));
+        @SuppressWarnings("unchecked")
+        Map<String, Object> data = (Map<String, Object>) target;
+        assertThat(data, notNullValue());
+        assertThat(data.get("ip"), equalTo(ip));
+        assertThat(data.get("city_name"), equalTo("Homestead"));
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> location = (Map<String, Object>) data.get("location");
+        assertThat(location, notNullValue());
+        assertThat(location, hasEntry("lat", 25.4573));
+        assertThat(location, hasEntry("lon", -80.4572));
+    }
+
+    public void testIpinfoGeolocationWithFlexibleFieldAccessMode() throws Exception {
+        String ip = "72.20.12.220";
+        GeoIpProcessor processor = new GeoIpProcessor(
+            IP_LOCATION_TYPE,
+            randomAlphaOfLength(10),
+            null,
+            "source_field",
+            loader("ipinfo/ip_geolocation_standard_sample.mmdb"),
+            () -> true,
+            "my.geo",
+            getIpinfoGeolocationLookup(),
+            false,
+            false,
+            "filename"
+        );
+
+        Map<String, Object> document = new HashMap<>();
+        document.put("source_field", ip);
+        IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random(), document);
+
+        // Execute with flexible field access mode
+        ingestDocument = runWithAccessPattern(FLEXIBLE, ingestDocument, processor);
+
+        // Verify that individual dotted fields were created (check in source map directly)
+        Map<String, Object> sourceAndMetadata = ingestDocument.getSourceAndMetadata();
+        assertThat(sourceAndMetadata.containsKey("my.geo.ip"), is(true));
+        assertThat(sourceAndMetadata.get("my.geo.ip"), equalTo(ip));
+        assertThat(sourceAndMetadata.containsKey("my.geo.city_name"), is(true));
+        assertThat(sourceAndMetadata.get("my.geo.city_name"), equalTo("Chicago"));
+        assertThat(sourceAndMetadata.containsKey("my.geo.country_iso_code"), is(true));
+
+        // Verify that location is written as an array [lon, lat] in flexible mode
+        assertThat(sourceAndMetadata.containsKey("my.geo.location"), is(true));
+        Object location = sourceAndMetadata.get("my.geo.location");
+        assertThat(location, instanceOf(List.class));
+        @SuppressWarnings("unchecked")
+        List<Double> locationArray = (List<Double>) location;
+        assertThat(locationArray.size(), equalTo(2));
+        // Verify longitude and latitude are present in the array
+        assertThat(locationArray.get(0), notNullValue()); // longitude
+        assertThat(locationArray.get(1), notNullValue()); // latitude
```

Comment:
Is there a reason you used `notNullValue` here instead of the actual resolved longitude and latitude (to ensure they are in the right order)?